content/en/blog/_posts/2023-12-13-Canary-Deployment-With-Kubernetes/index.md

```diff
@@ -0,0 +1,282 @@
+---
+layout: blog
+title: "A Comprehensive Guide to Canary Deployment with Kubernetes"
+date: 2023-12-13
+slug: canary-deployment-with-kubernetes
+evergreen: true
+---
+
+**Author**: Mohammed Affan
+
+## Introduction:
+
+Welcome to a comprehensive guide to canary deployment with kubernetes.
+In modern software development getting the quick feedback form the users
+is crucial for delivering a high quality product and implementing agile
+delivery. Developers try to get feedback from the users without ruining the
+user experience. That's where canary deployment comes into the picture.
+
+## Prerequisites:
+
+- You need to have a Kubernetes cluster (minikube would also work).
+- You also need to have `kubectl` installed on the local machine - see [install tools](/docs/tasks/tools/#kubectl).
+
```

Comment:
```suggestion
- You also need to have a basic understanding of service mesh and some sort of familiarity with Istio.

```
I think reader needs to have some sort of prior familiarity with Istio for following this blog.

Reply:
Thanks for the review.

However I think a beginner can also follow this guide with no familiarity with Istio at all. A user will get familiar with Istio in the blog itself.